### PR TITLE
compile fixes

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Fixes some compile warnings resulting in failing github actions.

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- /rebase will try to rebase the PR on the current develop branch.
- /reformat (experimental) applies the clang-format style changes as additional commit
- setting the label NoJenkins will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)

### CODE REVIEW

File: src/openms_gui/source/VISUAL/SpectraTreeTab.cpp [hunk 1]
```diff
@@ -187,11 +187,8 @@ namespace OpenMS
     spectra_treewidget_->setTreePosition(layer_->getCurrentSpectrumIndex());
   }
 
-  void SpectraTreeTab::itemSelectionChange_(QTreeWidgetItem* current, QTreeWidgetItem* previous)
+  void SpectraTreeTab::itemSelectionChange_(QTreeWidgetItem* current, QTreeWidgetItem*)
```
Comment (jpfeuffer) [inline src/openms_gui/source/VISUAL/SpectraTreeTab.cpp:190]:
```suggestion
  void SpectraTreeTab::itemSelectionChange_(QTreeWidgetItem* current, QTreeWidgetItem* /*previous*/)
```

File: src/openms_gui/source/VISUAL/SpectraIDViewTab.cpp [hunk 2]
```diff
@@ -994,9 +993,8 @@ namespace OpenMS
     }
   }
 
-  void SpectraIDViewTab::updatedSingleProteinCell_(QTableWidgetItem* item)
-  {
-    
+  void SpectraIDViewTab::updatedSingleProteinCell_(QTableWidgetItem*)
```
Comment (jpfeuffer) [inline src/openms_gui/source/VISUAL/SpectraIDViewTab.cpp:996]:
```suggestion
  void SpectraIDViewTab::updatedSingleProteinCell_(QTableWidgetItem* /*item*/)
```

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
I would just comment out unused vars

**Comment by jpfeuffer** [comment]
This broke the TOPPView_test

**Comment by timosachsenberg** [comment]
why? are you sure that this is related to https://github.com/OpenMS/OpenMS/pull/5764/files

**Pull request merged**